core/tests/test_apps.bitcoin.signtx_decred.py

```diff
@@ -0,0 +1,368 @@
+            helpers.UiConfirmOutput(out1, coin_decred),
+            True,
+            helpers.UiConfirmTotal(380000 + 10000, 10000, coin_decred),
+            True,
+            TxRequest(
+                request_type=TXINPUT,
+                details=TxRequestDetailsType(request_index=0, tx_hash=None),
+                serialized=EMPTY_SERIALIZED,
+            ),
+            TxAckInput(tx=TxAckInputWrapper(input=inp1)),
+            TxRequest(
+                request_type=TXMETA,
+                details=TxRequestDetailsType(
+                    request_index=None,
+                    tx_hash=unhexlify(
+                        "df8f9cf58455e8aa22d7f7be09d7877f7a0a698da7695152374c057a3047c24a"
+                    ),
+                ),
+                serialized=EMPTY_SERIALIZED,
+            ),
+            TxAckPrevMeta(tx=ptx1),
+            TxRequest(
+                request_type=TXINPUT,
+                details=TxRequestDetailsType(
+                    request_index=0,
+                    tx_hash=unhexlify(
+                        "df8f9cf58455e8aa22d7f7be09d7877f7a0a698da7695152374c057a3047c24a"
+                    ),
+                ),
+                serialized=EMPTY_SERIALIZED,
+            ),
+            TxAckPrevInput(tx=TxAckPrevInputWrapper(input=pinp1)),
+            TxRequest(
+                request_type=TXINPUT,
+                details=TxRequestDetailsType(
+                    request_index=1,
+                    tx_hash=unhexlify(
+                        "df8f9cf58455e8aa22d7f7be09d7877f7a0a698da7695152374c057a3047c24a"
+                    ),
+                ),
+                serialized=EMPTY_SERIALIZED,
+            ),
+            TxAckPrevInput(tx=TxAckPrevInputWrapper(input=pinp2)),
+            TxRequest(
+                request_type=TXOUTPUT,
+                details=TxRequestDetailsType(
+                    request_index=0,
+                    tx_hash=unhexlify(
+                        "df8f9cf58455e8aa22d7f7be09d7877f7a0a698da7695152374c057a3047c24a"
+                    ),
+                ),
+                serialized=EMPTY_SERIALIZED,
+            ),
+            TxAckPrevOutput(tx=TxAckPrevOutputWrapper(output=pout1)),
+            TxRequest(
+                request_type=TXINPUT,
+                details=TxRequestDetailsType(request_index=0, tx_hash=None),
+                serialized=EMPTY_SERIALIZED,
+            ),
+            TxAckInput(tx=TxAckInputWrapper(input=inp1)),
+            TxRequest(
+                request_type=TXFINISHED,
+                details=TxRequestDetailsType(request_index=None, tx_hash=None),
+                serialized=TxRequestSerializedType(
+                    signature_index=0,
+                    signature=unhexlify(
+                        "3044022078a5c388838796562eb9dad176b00e6d9425bc360083f633a14948685ca8a5ce02202a1b49cd44104a9d40aee8f988281a8aac94a497b5bc7337c77cc7ddbab16f23"
+                    ),
```

Comment:
Regarding https://github.com/trezor/trezor-firmware/pull/1249/files#r528434214:
```suggestion
                    ),
                    serialized_tx=bytearray(),
```

Reply:
This alone does not seem to solve the problem for me. It's like there is completely random data in res.serialized.serialized_tx.

Reply:
I think this fixes the issue which you were trying to fix using:
```
                if not res.serialized.serialized_tx:
                    res.serialized.serialized_tx = bytearray(0)
                if not response.serialized.serialized_tx:
                    response.serialized.serialized_tx = bytearray(0)
```

As for the data in the `res.serialized.serialized_tx`, that's where the signer progressively writes pieces of the serialized transaction. So if you were to concatenate all of the returned strings, then the result should be the expected transaction. If that's not the case then the logical conclusion would be that there is something wrong with the input or there is a bug in `decred.py`.

Reply:
Ok! I see now how it breaks down into parts of the transaction.